lib/chef/win32/api/file.rb

```diff
@@ -565,6 +600,21 @@ def retrieve_file_info(file_name)
           file_information
         end
 
+        def retrieve_file_version_info(file_name)
+          file_name = encode_path(file_name)
+          file_size = GetFileVersionInfoSizeW(file_name, "")
+          if file_size == 0
+            Chef::ReservedNames::Win32::Error.raise!
+          end
+          
+          version_info = ' '*file_size
```

Comment:
This makes me uneasy. While it may happen to work, there are better ways to do this. We should really be allocating a buffer and passing that through.